packages/roosterjs-editor-plugins/lib/plugins/Picker/PickerPlugin.ts

```diff
@@ -53,18 +53,65 @@ const UNIDENTIFIED_CODE = [0, 229];
  */
 export default class PickerPlugin<T extends PickerDataProvider = PickerDataProvider>
     implements EditorPlugin {
-    private editor: IEditor;
-    private eventHandledOnKeyDown: boolean;
-    private blockSuggestions: boolean;
-    private isSuggesting: boolean;
-    private lastKnownRange: Range;
+    private editor: IEditor | undefined = undefined;
+    private core: PickerPluginCore | undefined = undefined;
+    public dataProvider: T | undefined = undefined;
+    private pickerOptions: PickerPluginOptions | undefined = undefined;
+
+    constructor(dataProvider: T, pickerOptions: PickerPluginOptions) {
+        this.dataProvider = dataProvider;
+        this.pickerOptions = pickerOptions;
+    }
+
+    public willHandleEventExclusively(event: PluginEvent) {
+        return this.core ? this.core.willHandleEventExclusively(event) : false;
+    }
+
+    public initialize(editor: IEditor) {
+        this.editor = editor;
+        if (this.editor && this.dataProvider && this.pickerOptions) {
+            this.core = new PickerPluginCore(this.editor, this.dataProvider, this.pickerOptions);
+            this.core.initialize(editor);
```

Comment:
For the core class, when it is created, it already has everything (editor, provider, options) it needs, so you can let its ctor do initialize directly, no need to have a separate initialize function